### ISSUE TITLE
Requiring NPM

### ISSUE THREAD

**Issue by oskarkrawczyk** [KELiON/cerebro#57]
Are there any plans on not strictly requiring `npm`? Wanted to add some enhancements to cerebro, but `yarn install` fails with `AssertionError: Current node version is not supported for development` because (what I assume) NPM is required.

Would be cool to have the option to use the preferred package manager.

**Comment by Kageetai** [comment]
Yarn works fine for me. NPM itself is not required (as a package). This really sounds more like your version of Node is not correct or so...

**Comment by oskarkrawczyk** [comment]
I'd assume the node version is ok since I have almost the newest. Than again perhaps it doesn't work on the 7.x branch.

If you modify a bunch of scripts in `package.json` and replace with `yarn` all I get is:

```
[0] $ node -r babel-register server.js
[1] $ cross-env HOT=1 NODE_ENV=development ./node_modules/.bin/electron -r babel-register ./app/main.development
[1] fs.js:557
[1]   return binding.open(pathModule._makeLong(path), stringToFlags(flags), mode);
[1]                  ^
[1]
[1] Error: ENOENT: no such file or directory, open '/Users/oskarkrawczyk/Sites/cerebro/node_modules/electron-prebuilt/path.txt'
[1]     at Object.fs.openSync (fs.js:557:18)
[1]     at Object.fs.readFileSync (fs.js:467:33)
[1]     at Object.<anonymous> (/Users/oskarkrawczyk/Sites/cerebro/node_modules/electron-prebuilt/index.js:4:42)
[1]     at Module._compile (module.js:571:32)
[1]     at Object.Module._extensions..js (module.js:580:10)
[1]     at Module.load (module.js:488:32)
[1]     at tryModuleLoad (module.js:447:12)
[1]     at Function.Module._load (module.js:439:3)
[1]     at Module.require (module.js:498:17)
[1]     at require (internal/module.js:20:19)
[1] error Command failed with exit code 1.
[1] info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
[1] yarn run start-hot exited with code 1
--> Sending SIGTERM to other processes..
[0] Listening at http://localhost:3000
```

**Comment by KELiON** [comment]
@oskarkrawczyk what OS do you use for development? I had this problem on windows and I had to run `node install.js` from `./node_modules/electron-prebuilt` to fix it, but on linux/mac everything works well

**Comment by oskarkrawczyk** [comment]
@KELiON I'm on macOS 10.12.4b. I can confirm that running `node install.js` fixed the issue!

**Comment by maximbaz** [comment]
By the way, what is the purpose behind limiting to nodejs v6?

I've changed the `package.json` to have `"node": "7.x",` (because I have node v7) and it seems to be running just fine.

**Comment by KELiON** [comment]
@maximbaz actually it can be used with node 7 if you run `npm run rebuild` from `./app` directory to rebuild native-extensions. Otherwise, native extensions are built for node 7, not node 6, that is used by electron

**Comment by maximbaz** [comment]
Cool, in that case perhaps you should consider to loosen this node/npm version restriction, allow node 6+ to be used, just add the `cd app && npm run rebuild` to the `Install` section of your readme.

Personally this restriction (node == v6) is what held me back from setting up development environment, I have node v7 and I didn't want to spend time looking for a way to get node v6 on my machine. But like you said, node v7 can actually be used, and it works just fine.

**Comment by KELiON** [comment]
Ok, fixed in #67
